### DISCUSSION TITLE
Ubuntu 20.4 build EdgeTX Connecting to remote server

### DISCUSSION THREAD

**Discussion by flyb083** [EdgeTX/edgetx#3886] [Q&A]
I followed the steps in the WIKI to set up the EdgeTX environment and configure QT, but when I opened the CMakeLists.txt file in QT and selected the KIT of STM32, it was fine to build the firmware and exit normally, but the left project bar only showed all the header files, when I connected the development board through STLinkling, APPlication Output Loop output:

 [!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_JTAG_WRITEDEBUG_32BIT

I don't know how to solve the problem. Will you help me out? 
I don't usually upload pictures, so I don't know if this link will let you see the pictures

https://pan.baidu.com/s/1EdvlYGb9JvpisoRpEg8Nnw?pwd=yr1y

**Comment by flyb083** [comment]
Compile Output:
19:01:10: Running steps for project EdgeTX...
19:01:10: Starting: "/usr/bin/cmake" --build . --target firmware
[100%] Built target arm-none-eabi-configure
[  0%] Built target taranis_xbm
[ 13%] Built target taranis_fonts
[ 16%] Built target taranis_bitmaps
[ 19%] Built target stm32cube_ll
[ 21%] Built target stm32_stdperiph
[ 22%] Built target minimal_board_lib
[ 33%] Built target bootloader
[ 35%] Built target freertos
[ 37%] Built target board_lib
[100%] Built target firmware
[100%] Built target firmware
19:01:14: The process "/usr/bin/cmake" exited normally.
19:01:14: Elapsed time: 00:04.

**Comment by flyb083** [comment]
19:52:47: Starting /usr/bin/st-util --stlink_version=2 --listen_port=4242 --verbose=0...
19:52:47: Debugging starts
st-util 1.6.0
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_DEBUG_RESETSYS
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_DEBUG_FORCEDEBUG
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_DEBUG_RESETSYS
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_JTAG_WRITEDEBUG_32BIT
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_JTAG_READDEBUG_32BIT
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_JTAG_READDEBUG_32BIT
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_JTAG_WRITEDEBUG_32BIT
[!] send_recv send request failed: LIBUSB_ERROR_IO
[!] send_recv STLINK_JTAG_WRITEDEBUG_32BIT

**Comment by flyb083** [comment]
@rotorman

**Reply by rotorman** [reply]
Can you try just to drag and drop the image file you wish to attach to GitHub - the link above is not showing an image.

**Reply by flyb083** [reply]
Thank you for your reply, I tried this, but it showed an error:
Something went really wrong, and we can't process that file. Try again.

**Reply by flyb083** [reply]
![Uploading proble,.png…]()

**Reply by pfeerick** [reply]
This is the image from that link:
![12a79b546reff85c6820cdefdc8caae0](https://github.com/EdgeTX/edgetx/assets/5500713/1e402632-5394-44ab-b69e-fa0dc2a50019)

**Reply by flyb083** [reply]
that's right !Thank you very much! Can you help me see what the problem is?

**Reply by rotorman** [reply]
The in-circuit-debug steps with Qt Creator in the Wiki are not up to date anymore. Can you try with Segger Ozone instead?

**Reply by flyb083** [reply]
Thank you for your suggestion, but I still want to edit source code with Qt Creator later. Could you please tell me more information about configuring the EDGETX environment with QT?
